unit_test/graph/Test_Graph_graph_color_distance2.hpp

```diff
@@ -61,231 +61,291 @@ using namespace KokkosGraph::Experimental;
-
-    //Note that MATRIX_SQUARED cannot be used here, because SPGEMM doesn't accept out-of-bounds indices
-    run_graphcolor_d2<crsMat_type, device>(input_mat, input_mat,
-        {COLORING_D2_SERIAL, COLORING_D2, COLORING_D2_VB, COLORING_D2_VB_BIT, COLORING_D2_VB_BIT_EF, COLORING_D2_NB_BIT});
 }
+#define DO_DEPRECATED_TEST(SCALAR, ORDINAL, OFFSET, DEVICE) \
+      test_old_d2<SCALAR, ORDINAL, OFFSET, DEVICE>(2000, 4000, 3000 * 20, 800, 10); \
+      test_old_d2<SCALAR, ORDINAL, OFFSET, DEVICE>(4000, 2000, 3000 * 20, 800, 10);
+#else
+#define DO_DEPRECATED_TEST(SCALAR, ORDINAL, OFFSET, DEVICE)
+#endif
 
-#define EXECUTE_TEST(SCALAR, ORDINAL, OFFSET, DEVICE)                                           \
-    TEST_F(TestCategory, graph##_##graph_color_d2_symmetric##_##SCALAR##_##ORDINAL##_##OFFSET##_##DEVICE) \
-    {                                                                                           \
-      test_d2_symmetric<SCALAR, ORDINAL, OFFSET, DEVICE>(5000, 5000 * 30, 200, 10);          \
-      test_d2_symmetric<SCALAR, ORDINAL, OFFSET, DEVICE>(5000, 5000 * 5, 100, 10);          \
+#define EXECUTE_TEST(SCALAR, ORDINAL, OFFSET, DEVICE) \
+    TEST_F(TestCategory, graph##_##graph_color_distance2##_##SCALAR##_##ORDINAL##_##OFFSET##_##DEVICE) \
+    { \
+      test_dist2_coloring<SCALAR, ORDINAL, OFFSET, DEVICE>(5000, 5000 * 20, 1000, 10); \
+      test_dist2_coloring<SCALAR, ORDINAL, OFFSET, DEVICE>(50, 50 * 10, 40, 10); \
+    } \
+    TEST_F(TestCategory, graph##_##graph_color_deprecated_distance2##_##SCALAR##_##ORDINAL##_##OFFSET##_##DEVICE) \
+    { \
+      DO_DEPRECATED_TEST(SCALAR, ORDINAL, OFFSET, DEVICE) \
     } \
-    TEST_F(TestCategory, graph##_##graph_color_d2_asymmetric##_##SCALAR##_##ORDINAL##_##OFFSET##_##DEVICE) \
-    {                                                                                           \
-      test_d2_asymmetric<SCALAR, ORDINAL, OFFSET, DEVICE>(5000, 6000, 5000 * 30, 200, 10);          \
-      test_d2_asymmetric<SCALAR, ORDINAL, OFFSET, DEVICE>(5000, 6000, 5000 * 5, 100, 10);          \
+    TEST_F(TestCategory, graph##_##graph_color_bipartite_row##_##SCALAR##_##ORDINAL##_##OFFSET##_##DEVICE) \
+    { \
+      test_bipartite<SCALAR, ORDINAL, OFFSET, DEVICE>(2000, 4000, 3000 * 20, 800, 10, true); \
+      test_bipartite<SCALAR, ORDINAL, OFFSET, DEVICE>(4000, 2000, 3000 * 20, 800, 10, true); \
     } \
-    TEST_F(TestCategory, graph##_##graph_color_d2_filtering##_##SCALAR##_##ORDINAL##_##OFFSET##_##DEVICE) \
-    {                                                                                           \
-      test_d2_filtering<SCALAR, ORDINAL, OFFSET, DEVICE>(5000, 7000, 5000 * 10, 200, 10); \
+    TEST_F(TestCategory, graph##_##graph_color_bipartite_col##_##SCALAR##_##ORDINAL##_##OFFSET##_##DEVICE) \
+    { \
+      test_bipartite<SCALAR, ORDINAL, OFFSET, DEVICE>(2000, 4000, 3000 * 20, 800, 10, false); \
```

Comment:
What happens if we give a square symmetric matrix here ?

Reply:
@srajama1 It will still work, but it will compute the explicit transpose internally which is just inefficient. I will add a symmetric bipartite (i.e. no difference for coloring rows vs. columns) test.